File: src/test/scala/com/github/loadtest4j/drivers/gatling/DriverRequests.scala
package com.github.loadtest4j.drivers.gatling

import com.github.loadtest4j.loadtest4j.DriverRequest
import java.util.Collections

import scala.collection.JavaConverters

object DriverRequests {
    def get(path: String): DriverRequest = {
        new DriverRequest("", Collections.emptyMap(), "GET", path, Collections.emptyMap())
    }

  def getWithQueryParams(path: String, queryParams: Map[String, String]): DriverRequest = {
    val javaQueryParams = JavaConverters.mapAsJavaMap(queryParams)
    new DriverRequest("", Collections.emptyMap(), "GET", path, javaQueryParams)
  }

    def post(path: String, body: String, headers: Map[String, String]): DriverRequest = {
        val javaHeaders = JavaConverters.mapAsJavaMap(headers)
        new DriverRequest(body, javaHeaders, "POST", path, Collections.emptyMap())
    }
}

File: src/test/java/com/github/loadtest4j/drivers/gatling/DriverRequests.java
package com.github.loadtest4j.drivers.gatling;

import com.github.loadtest4j.loadtest4j.driver.DriverRequest;

import java.util.Collections;
import java.util.Map;

class DriverRequests {
    protected static DriverRequest get(String path) {
        return new DriverRequest("", Collections.emptyMap(), "GET", path, Collections.emptyMap());
    }

    protected static DriverRequest getWithQueryParams(String path, Map<String, String> queryParams) {
        return new DriverRequest("", Collections.emptyMap(), "GET", path, queryParams);
    }

    protected static DriverRequest post(String path, String body, Map<String, String> headers) {
        return new DriverRequest(body, headers, "POST", path, Collections.emptyMap());
    }
}
